mllib/src/main/scala/org/apache/spark/ml/util/MLUtils.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ml.util
+
+import org.apache.spark.SparkConf
+import org.apache.spark.ml.tree.impl._
+import org.apache.spark.util.Utils
+
+private[spark] object MLUtils {
+
+  private[this] var kryoRegistered: Boolean = false
+
+  def registerKryoClasses(conf: SparkConf): Unit = {
```

Comment:
synchronized?
This is the only part that worries me. I don't know how stable the class names are or if there are other implications of registering them (probably not). Do these really have to get serialized to make it work?

Reply:
I think mark it synchronized is good, here follows same impl in `GraphXUtils` and `SquaredEuclideanSilhouette`. I think we can make them `synchronized` in the future.

> Do these really have to get serialized to make it work?

Since current `TreePoint` was registered in `object KryoSerializer`, so I tried to also register `TreePoint[B]`, otherwise, I may need to swith to a new treepoint class.